### ISSUE TITLE
Breaking change and regression with 4.4.2

### ISSUE THREAD

**Issue by valeriangalliat** [vitaly-t/pg-promise#158]
Version 4.4.2 introduced [configuration path support](https://github.com/vitaly-t/pg-promise/commit/9ea93854a791f8f9289e90b71b0bfb86f67a338f) but it broke two things (with just a patch bump):

---

Can't use local database name as connection string anymore; I used to do `require('pg-promise')()('somedb')` but now it tries to require `somedb` as configuration file. I don't know if this module uses SemVer, but if so, it should probably have been introduced with a major version bump.

I have nothing against doing `postgres://somedb` as of now, but the update broke existing code, with a very misleading error message (that I include to keep track, might help other people having the same problem):

```
module.js:442
    throw err;
    ^

Error: Cannot find module '/tmp/somedb'
    at Function.Module._resolveFilename (module.js:440:15)
    at Function.Module._load (module.js:388:25)
    at Module.require (module.js:468:17)
    at require (internal/module.js:20:19)
    at parseConnection (/tmp/node_modules/pg-promise/lib/index.js:235:18)
    at inst (/private/tmp/node_modules/pg-promise/lib/index.js:181:29)
    at Object.<anonymous> (/tmp/test.js:2:24)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:458:32)
```

---

Also, [this line](https://github.com/vitaly-t/pg-promise/commit/9ea93854a791f8f9289e90b71b0bfb86f67a338f#diff-eae83c5838640e1c85538d8c9fbe35e3R109) requires `process.argv[1]` to be defined, otherwise we end up with the following error:

```
TypeError: Path must be a string. Received undefined
    at assertPath (path.js:7:11)
    at Object.dirname (path.js:1324:5)
    at Object.<anonymous> (/tmp/node_modules/pg-promise/lib/utils/index.js:128:21)
    at Module._compile (module.js:541:32)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3)
    at Module.require (module.js:468:17)
    at require (internal/module.js:20:19)
```

And `process.argv[1]` is not defined when we run code directly from the Node interpreter. This breaks at `require` time so it means we can't require `pg-promise` nor anything depending on it from the Node interpreter.

---

What's pg-promise position on SemVer? For the `process.argv[1]` problem, I made a PR (#157) to fallback to `process.cwd()`.

**Comment by vitaly-t** [comment]
I wasn't aware anybody was using it with `pgp('somedb')`. This isn't really part of any documentation, if if it worked - it was working unofficially.

**Comment by valeriangalliat** [comment]
`pg-connection-string` parses it as database name:

```js
require('pg-connection-string').parse('foo')
{ port: null,
  host: null,
  database: 'foo',
  user: '',
  password: '' }
```

That's because it's treated as a URL and `foo` is an URL with just a pathname as parsed by `url.parse`. It's not explicitly documented but that's a natural consequence of the connection string being treated as an URL.

I always used this when using `pg` and `pg-promise` before, like I would do `psql foo` to connect to local `foo` database (I found it intuitive at least, but if I'm the only one doing this that's probably not worth supporting).

But note this change also removes support from socket connection strings [documented here](https://github.com/brianc/node-postgres/wiki/pg#parameters).

Having different connection interface with `pg`, even though it's documented, can break the user expectation, especially when moving from `pg` to `pg-promise`. While the connection parameter is properly documented by `pg-promise`, I'd think it's better for an API design point of view to have `pg-promise` always pass the connection string/object raw to `pg` and let it handle it.

Feel free to close this issue if you don't want to maintain 1:1 compatibility with `pg` for the connection parameter, I just wanted to make sure it's intended. 👍

**Comment by vitaly-t** [comment]
>But note this change also removes support from socket connection strings

That's a big Ops! :smile: 

I'm gonna be making some changes now for the next 4.5 release, which will also include `node-postgres` 5.0.0 dependency which was just released.

Thank you for the findings!

**Comment by vitaly-t** [comment]
I am removing the entire Configuration File feature from 4.5.0

**Comment by vitaly-t** [comment]
Fixed in [4.5.0](https://github.com/vitaly-t/pg-promise/releases/tag/v.4.5.0).

**Comment by valeriangalliat** [comment]
Thanks a lot!
